toolchain/gcc-14.2.0-13-arm-multilib.patch

```diff
@@ -0,0 +1,208 @@
+-MULTILIB_REQUIRED	=
+-
+-MULTILIB_OPTIONS	+= mfloat-abi=hard
+-MULTILIB_DIRNAMES	+= fpu
+-MULTILIB_MATCHES	+= mfloat-abi?hard=mhard-float
+-
+-MULTILIB_OPTIONS	+= mno-thumb-interwork/mthumb-interwork
+-MULTILIB_DIRNAMES	+= normal interwork
++# Add phoenix options to reuses and requires.
++
++s-mlib: $(srcdir)/config/arm/t-phoenix
++
++# Find all RM archs.
++RM_ARCHS			:= $(subst /, ,march=armv7-r+fp.sp/$(MULTI_ARCH_OPTS_RM))
++# Create matches for requires and reuses.
++RM_MATCHES_REQ		:= $(foreach ARCH, $(RM_ARCHS), mthumb/$(ARCH)% marm/$(ARCH)%)
++RM_MATCHES_REUSE	:= $(foreach TARGET, $(RM_MATCHES_REQ), $(subst =,.,$(subst .,\.,$(RM_MATCHES_REQ))))
++
++# NOTE: Multilib options in MULTILIB_REQUIRED must be in the same order as they were added to the MULTILIB_OPTIONS variable.
++# So to easily add custom options at the end of REQUIRE we have to add them at the end of OPTIONS and DIRS.
++MULTILIB_OPTIONS	+= fPIC mno-pic-data-is-text-relative/mfdpic
++MULTILIB_DIRNAMES	+= pic no-pditr fdpic
++
++MULTILIB_MATCHES	+= fPIC=fpic fPIC=fPIE fPIC=fpie
++
++PHOENIX_OPTS		:= /fPIC/mno-pic-data-is-text-relative
++PHOENIX_OPTS		+= /fPIC/mfdpic
++
++# Finally, add Phoenix options to RM multilibs.
++# Keep the orignal requires because of kernel and plo.
++
++RM_REQUIRED			:= $(filter $(RM_MATCHES_REQ),$(MULTILIB_REQUIRED))
++
++RM_REUSE			:= $(filter $(RM_MATCHES_REUSE),$(MULTILIB_REUSE))
++
++MULTILIB_REQUIRED	+= $(foreach REQ, $(RM_REQUIRED), \
++                    $(foreach OPT, $(PHOENIX_OPTS), $(REQ)$(OPT)))
++
++MULTILIB_REUSE		+= $(foreach REUSE, $(RM_REUSE), \
++                    $(foreach OPT, $(PHOENIX_OPTS), $(subst =,$(OPT)=,$(REUSE))$(OPT)))
++
++# FIXME: Disable FDPIC Thumb-1 targets as binutils do not support FDPIC for them.
++# https://github.com/bminor/binutils-gdb/blob/67bffda3bff903a78874c303a3d7823899bd555e/bfd/elf32-arm.c#L9838
++THUMB1_ARCHS		:= march=armv6s-m march=armv8-m.base
++
++THUMB1_REUSE_MATCHES+= $(foreach ARCH, $(THUMB1_ARCHS), $(subst =,.,$(subst .,\.,mthumb/$(ARCH)%fdpic marm/$(ARCH)%fdpic)))
++
++MULTILIB_EXCEPTIONS	+= $(foreach ARCH, $(THUMB1_ARCHS), *$(ARCH)*/*mfdpic*)
++
++MULTILIB_REUSE		:= $(filter-out $(THUMB1_REUSE_MATCHES), $(MULTILIB_REUSE))
+diff --git a/libgcc/config.host b/libgcc/config.host
+--- a/libgcc/config.host
++++ b/libgcc/config.host
+@@ -563,7 +563,7 @@ arm*-*-uclinux*)		# ARM ucLinux
+ 	extra_parts="$extra_parts crti.o crtn.o"
+ 	;;
+ arm*-*-phoenix*)
+-	tmake_file="t-hardfp t-softfp arm/t-arm arm/t-elf arm/t-softfp arm/t-phoenix"
++	tmake_file="t-hardfp t-softfp arm/t-arm arm/t-elf arm/t-softfp"
+ 	tmake_file="${tmake_file} arm/t-bpabi"
+ 	tmake_file="${tmake_file} t-crtstuff-pic t-libgcc-pic"
+ 	tm_file="$tm_file arm/bpabi-lib.h"
+diff --git a/libgcc/config/arm/t-phoenix b/libgcc/config/arm/t-phoenix
+deleted file mode 100644
+index 9b6aafc7a51..00000000000
+--- a/libgcc/config/arm/t-phoenix
++++ /dev/null
+@@ -1,13 +0,0 @@
+-# PhoenixRTOS uses relocatable ELFs on NOMMU platforms
```

Comment:
just making sure: it's safe to remove this patch because we will add `-mno-pic-data-is-text-relative` as a part of multilib build?